src/handler.rs

```diff
@@ -130,14 +135,28 @@ impl RequestHandler {
                 remote_addr_str.push_str(&client_ip_address.to_string())
             }
         }
-        tracing::info!(
-            "incoming request: method={} uri={}{}",
-            method,
-            uri,
-            remote_addr_str,
-        );
+
+        if health_request {
+            tracing::debug!(
+                "incoming request: method={} uri={}{}",
+                method,
+                uri,
+                remote_addr_str,
+            );
+        } else {
+            tracing::info!(
+                "incoming request: method={} uri={}{}",
+                method,
+                uri,
+                remote_addr_str,
+            );
+        }
 
         async move {
+            if health_request {
+                return Ok(Response::new(Body::from("OK")));
+            }
```

Comment:
Here are some minor changes to do:

- Append a `content-type: text/html`  header.
- Add support for `HEAD` requests as well and skip the body only in this case.